tests/test_scoring.py

```diff
@@ -0,0 +1,80 @@
+"""Smoke-tests for fast scoring."""
+
+from __future__ import annotations
+
+import pytest
+import torch
+
+from pykeen.models import DistMult, ERModel
+from pykeen.models.scoring import Batch, Scorer
+from pykeen.triples import KGInfo
+from pykeen.typing import LongTensor
+
+
+@pytest.fixture()
+def generator() -> torch.Generator:
+    """Build a random number generator with fixed seed."""
+    return torch.manual_seed(42)
+
+
+def make_tensor(size: tuple[int, ...] | None, num: int, generator: torch.Generator) -> LongTensor | None:
+    """Create a tensor with the given size."""
+    if size is None:
+        return None
+    return torch.randint(num, size=size, generator=generator)
+
+
+NUM_ENTITIES = 7
+NUM_RELATIONS = 5
+
+
+@pytest.fixture()
+def model() -> ERModel:
+    """Build a test model."""
+    return DistMult(
+        triples_factory=KGInfo(num_entities=NUM_ENTITIES, num_relations=NUM_RELATIONS, create_inverse_triples=False),
+        embedding_dim=8,
+    )
+
+
+B = 2
+K = 3
+
+
+@pytest.mark.parametrize(
```

Comment:
i think it's great how elegantly this generalizes all of the different scoring functions from the ERModel (besides, inverses, which still seem like a problem 😅 )